packages/jss/src/utils/escape.js

```diff
@@ -1,11 +1,4 @@
 const escapeRegex = /([[\].#*$><+~=|^:(),"'`\s])/g
+const nativeEscape = typeof CSS !== 'undefined' && CSS.escape
 
-export default str => {
-  // We don't need to escape it in production, because we are not using user's
-  // input for selectors, we are generating a valid selector.
-  if (process.env.NODE_ENV === 'production') return str
-
-  const nativeEscape = typeof CSS !== 'undefined' && CSS.escape
-
-  return nativeEscape ? nativeEscape(str) : str.replace(escapeRegex, '\\$1')
-}
+export default str => (nativeEscape ? nativeEscape(str) : str.replace(escapeRegex, '\\$1'))
```

Comment:
We will always need to escape the string now because the user might not minify the string

We could also pass here the sheet options and check if the user wants to minify the selector